main.nf

```diff
@@ -164,6 +141,20 @@ if( !(workflow.runName ==~ /[a-z]+_[a-z]+/) ){
   custom_runName = workflow.runName
 }
 
+if( workflow.profile == 'awsbatch') {
+  // AWSBatch sanity checking
+  if (!params.awsqueue || !params.awsregion) exit 1, "Specify correct --awsqueue and --awsregion parameters on AWSBatch!"
+  // Check outdir paths to be S3 buckets if running on AWSBatch
+  // related: https://github.com/nextflow-io/nextflow/issues/813
+  if (!params.outdir.startsWith('s3:')) exit 1, "Outdir not on S3 - specify S3 Bucket to run on AWSBatch!"
+  // Prevent trace files to be stored on S3 since S3 does not support rolling files.
+  if (workflow.tracedir.startsWith('s3:')) exit 1, "Specify a local tracedir or run without trace! S3 cannot be used for tracefiles."
+}
+
+// Stage config files
+ch_multiqc_config = Channel.fromPath(params.multiqc_config)
```

Comment:
```suggestion
ch_multiqc_config = Channel.fromPath(params.multiqc_config, checkIfExists: true)
```

Reply:
You can add `checkIfExists: true` anywhere where you are using `file()` or `Channel.fromPath()`.

Reply:
You have another line in `main.nf` that is reading the same file.
https://github.com/nf-core/smrnaseq/blob/9bbf1ffe410fcdee983a9eedeec7c63caccc7f7b/main.nf#L135

Reply:
ok, I added to two more places I found to be similar. :)

Reply:
> You have another line in `main.nf` that is reading the same file.
> https://github.com/nf-core/smrnaseq/blob/9bbf1ffe410fcdee983a9eedeec7c63caccc7f7b/main.nf#L135

You have this:
`multiqc_config = file(params.multiqc_config, checkIfExists: true)`

and this:
`ch_multiqc_config = Channel.fromPath(params.multiqc_config, checkIfExists: true)`

Reply:
Only need one 👍

Reply:
> You have another line in `main.nf` that is reading the same file.
> https://github.com/nf-core/smrnaseq/blob/9bbf1ffe410fcdee983a9eedeec7c63caccc7f7b/main.nf#L135

You still have two instances of this:
https://github.com/nf-core/smrnaseq/blob/b5da260cfc809c0d05ab14f57ba35a61ae29cb1d/main.nf#L138

https://github.com/nf-core/smrnaseq/blob/b5da260cfc809c0d05ab14f57ba35a61ae29cb1d/main.nf#L158

Reply:
sorry about that.